src/components/ui/TopCornerMenuService.vue

```diff
@@ -0,0 +1,154 @@
+            @action="serviceStop"
+            @close="showStopDialog = false" />
+    </v-list-item>
+</template>
+<script lang="ts">
+import Component from 'vue-class-component'
+import { Mixins, Prop } from 'vue-property-decorator'
+import BaseMixin from '@/components/mixins/base'
+import { mdiPlay, mdiRestart, mdiStop } from '@mdi/js'
+import { capitalize } from '@/plugins/helpers'
+import ServiceMixins from '@/components/mixins/services'
+
+@Component({})
+export default class TopCornerMenuService extends Mixins(BaseMixin, ServiceMixins) {
+    mdiPlay = mdiPlay
+    mdiRestart = mdiRestart
+    mdiStop = mdiStop
+
+    @Prop({ type: String, required: true }) service!: string
+
+    showRestartDialog = false
+    showStopDialog = false
+
+    get name() {
+        if (this.hideOtherInstances && this.service === this.klipperInstance) return 'Klipper'
+        if (this.hideOtherInstances && this.service === this.moonrakerInstance) return 'Moonraker'
+
+        return capitalize(this.service)
+    }
+
+    get service_states() {
+        return this.$store.state.server.system_info?.service_state ?? {}
+    }
+
+    get state() {
+        if (this.service in this.service_states) return this.service_states[this.service].active_state
+
+        return null
+    }
+
+    get subState() {
+        if (this.service in this.service_states) return this.service_states[this.service].sub_state
+
+        return null
+    }
+
+    get dialogRestartTitle() {
+        if (this.service === this.klipperInstance)
+            return this.$t('App.TopCornerMenu.ConfirmationDialog.Title.KlipperRestart')
+
+        return this.$t('App.TopCornerMenu.ConfirmationDialog.Title.ServiceRestart')
+    }
+
+    get dialogStopTitle() {
+        return this.$t('App.TopCornerMenu.ConfirmationDialog.Title.ServiceStop')
+    }
+
+    get dialogRestartDescription() {
+        if (this.service === this.klipperInstance)
+            return this.$t('App.TopCornerMenu.ConfirmationDialog.Description.KlipperRestart')
+
+        return this.$t('App.TopCornerMenu.ConfirmationDialog.Description.ServiceRestart')
+    }
+
+    get dialogStopDescription() {
+        if (this.service === this.klipperInstance)
+            return this.$t('App.TopCornerMenu.ConfirmationDialog.Description.KlipperStop')
+
+        return this.$t('App.TopCornerMenu.ConfirmationDialog.Description.ServiceStop')
+    }
+
+    get disableStopButton() {
+        return this.state === 'inactive' || this.service === this.moonrakerInstance
+    }
+
+    get styleStopButton() {
+        return this.service === this.moonrakerInstance ? 'visibility: hidden;' : ''
+    }
+
+    clickStart() {
+        this.$socket.emit('machine.services.start', { service: this.service })
+        this.closeMenu()
+    }
+
+    clickRestart() {
+        if (this.printerIsPrinting) {
+            this.showRestartDialog = true
+            return
+        }
+
+        this.serviceRestart()
```

Comment:
calling `serviceRestart()` here skips the confirmation dialog and restarts the service immediately. 
Show the dialog instead. The dialog takes care of the restart then.
```suggestion
        this.showRestartDialog = true
```